apps/emqx/src/emqx_persistent_session_ds_subs.erl

```diff
@@ -244,7 +246,14 @@ to_map(S) ->
 ) ->
     Acc.
 fold(Fun, Acc, S) ->
-    emqx_persistent_session_ds_state:fold_subscriptions(Fun, Acc, S).
+    emqx_persistent_session_ds_state:fold_subscriptions(
```

Comment:
Hiding data from a generic function like `fold` will lead to confusion. Filtering shared subs is better done by business logic.

Reply:
Do you mean it is the callers who should decide which subscriptions to filter?